src/main/java/ch/njol/skript/classes/ClassInfo.java

```diff
@@ -162,7 +167,34 @@ public ClassInfo<T> defaultExpression(final DefaultExpression<T> defaultExpressi
 		this.defaultExpression = defaultExpression;
 		return this;
 	}
-	
+
+
+	/**
+	 * Used for dynamically getting all the possible values of a class
+	 *
+	 * @param supplier The supplier of the values
+	 * @return This ClassInfo object
+	 * @see ClassInfo#supplier(T[])
+	 */
+	public ClassInfo<T> supplier(Supplier<T[]> supplier) {
```

Comment:
what do you think about supplying an `Iterator<T>` instead? this would help performance in loops of classinfos with many values as `ExprSets#iterator` can use that instead

Reply:
iterators can't be dynamic unlike suppliers, meaning it gives freedom to addon developers to supply their own dynamic or constant values and that's not really possible with iterators

Reply:
sorry, I wasnt clear. I mean a `Supplier<Iterator<T>>`

Reply:
So far looks good to me but are you planning to add this suggestion?

Reply:
> So far looks good to me but are you planning to add this suggestion?

yes, i just did :grin: